src/foam/box/SessionServerBox.java

```diff
@@ -68,22 +68,27 @@ public void send(Message msg) {
 
           if ( authenticate_ && ! auth.check(session.getContext(), "service." + spec.getName()) ) {
             Logger logger   = (Logger) x.get("logger");
-            logger.debug("missing permission", group.getId(), "service." + spec.getName());
+            logger.debug("missing permission", group != null ? group.getId() : "NO GROUP" , "service." + spec.getName());
             // msg.replyWithException(new NoPermissionException("No permission"));
             // return;
           }
 
+          // padding this cause if group is null this can cause an NPE
+          // technically the user shouldn't be created without a group
+          if (group != null) {
```

Comment:
How can this happen or why is this being done?

Reply:
This is exactly the problem we are trying to address.